File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.cpp
#include <vector>
#include <algorithm>
#include <numeric>
using namespace std;

class Solution {
public:
    int findGCD(vector<int>& nums) {
        int minVal = *min_element(nums.begin(), nums.end());
        int maxVal = *max_element(nums.begin(), nums.end());
        return gcd(minVal, maxVal);
    }
};

File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.java
import java.util.Arrays;

class Solution {
    public int findGCD(int[] nums) {
        int minVal = Arrays.stream(nums).min().getAsInt();
        int maxVal = Arrays.stream(nums).max().getAsInt();
        return gcd(minVal, maxVal);
    }
    
    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}